presto-main/src/main/java/com/facebook/presto/server/protocol/QueuedStatementResource.java

```diff
@@ -170,7 +174,7 @@ public Response postStatement(
             throw badRequest(BAD_REQUEST, "SQL statement is empty");
         }
 
-        SessionContext sessionContext = new HttpRequestSessionContext(servletRequest, sqlParserOptions);
+        SessionContext sessionContext = new HttpRequestSessionContext(servletRequest, sqlParserOptions, tracerProvider.getNewTracer());
```

Comment:
- Does tracer starts its timer when it is created? The reason for this question is that if it doesn't start when it is created, then we don't have to pass it in here... If it is, I agree we need to initialize it as early as possible. It would be good to add a comment to explain.
- Can we add a TODO to pass in `traceToken` into `getNewTracer`?

Reply:
Yes the trace starts its timer when it is created (by calling TracerProvider.getNewTracer()). I'll add a TODO here.